openedx/core/djangoapps/content/learning_sequences/views.py

```diff
@@ -154,36 +152,35 @@ def get(self, request, course_key_str, format=None):  # lint-amnesty, pylint: di
         """
         The CourseOutline, customized for a given user.
 
-        Currently restricted to global staff.
-
         TODO: Swagger docs of API. For an exemplar to imitate, see:
         https://github.com/edx/edx-platform/blob/master/lms/djangoapps/program_enrollments/rest_api/v1/views.py#L792-L820
         """
-        # Translate input params and do any substitutions...
+        # Translate input params and do course key validation (will cause HTTP
+        # 400 error if an invalid CourseKey was entered, instead of 404).
         course_key = self._validate_course_key(course_key_str)
         at_time = datetime.now(timezone.utc)
-        user = self._determine_user(request)
 
-        # Grab the user's outline and send our response...
+        if not can_call_public_api(request.user, course_key):
+            raise PermissionDenied()
+
         try:
-            user_course_outline_details = get_user_course_outline_details(course_key, user, at_time)
+            # Grab the user's outline and send our response...
+            outline_user = self._determine_user(request)
+            user_course_outline_details = get_user_course_outline_details(course_key, outline_user, at_time)
         except CourseOutlineData.DoesNotExist as does_not_exist_err:
             raise NotFound() from does_not_exist_err
 
         serializer = self.UserCourseOutlineDataSerializer(user_course_outline_details)
         return Response(serializer.data)
 
-    def _validate_course_key(self, course_key_str):  # lint-amnesty, pylint: disable=missing-function-docstring
+    def _validate_course_key(self, course_key_str):
+        """Validate the Course Key and raise a ValidationError if it fails."""
         try:
             course_key = CourseKey.from_string(course_key_str)
-        except InvalidKeyError:
-            raise serializers.ValidationError(  # lint-amnesty, pylint: disable=raise-missing-from
-                f"{course_key_str} is not a valid CourseKey"
-            )
+        except InvalidKeyError as err:
+            raise serializers.ValidationError(f"{course_key_str} is not a valid CourseKey") from err
         if course_key.deprecated:
-            raise serializers.ValidationError(
-                "Deprecated CourseKeys (Org/Course/Run) are not supported."
-            )
+            raise serializers.ValidationError("Deprecated CourseKeys (Org/Course/Run) are not supported.")
         return course_key
 
     def _determine_user(self, request):
```

Comment:
Nit: This method feels like something that should be more core to the platform. Maybe it even exists already?

Reply:
Discussions has basically identical wording, but I don't know as it's worth refactoring into a common message yet.